pkg/network/message.go

```diff
@@ -1,29 +1,52 @@
 package network
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
+	iocore "io"
```

Comment:
You can move compress/decompress functions into their own `compress.go` here and avoid doing this.